### PR TITLE
GEOMESA-3211 Add Geomesa UDF support in PySpark

### PR DESCRIPTION
**Pull request by mkearns-ccri**
* updates the geomesa pyspark documentation
* wraps geomesa nullableUDFs as UserDefinedFunctions
* adds utility functions for creating scala UDFs in pyspark
* adds python wrappers for the geomesa UDFs

### CODE REVIEW

**Comment by mkearns-ccri** [comment]
@elahrvivaz @ronq @jrs53 @fdfea

File: geomesa-spark/geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/jts/udf/GeomesaPysparkFunctions.scala [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  /* Geometric Cast Functions */
+  def st_castToPoint: UserDefinedFunction = udf(ST_CastToPoint)
+  def st_castToPolygon: UserDefinedFunction = udf(ST_CastToPolygon)
+  def st_castToLineString: UserDefinedFunction = udf(ST_CastToLineString)
+  def st_castToGeometry: UserDefinedFunction = udf(ST_CastToGeometry)
+  def st_byteArray: UserDefinedFunction = udf(ST_ByteArray)
+
+  /* Geometric Constructor Functions */
+  def st_geomFromGeoHash: UserDefinedFunction = udf(ST_GeomFromGeoHash)
+  def st_box2DFromGeoHash: UserDefinedFunction = udf(ST_GeomFromGeoHash)
```
Comment (elahrvivaz) [inline geomesa-spark/geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/jts/udf/GeomesaPysparkFunctions.scala:55]: just double checking this is the right delegate function?

Comment (elahrvivaz) [reply geomesa-spark/geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/jts/udf/GeomesaPysparkFunctions.scala:55]: seems like it is from the comment in functions.py

**Comment by elahrvivaz** [comment]
I don't really know python, but lgtm.

File: geomesa-spark/geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/jts/udf/GeomesaPysparkFunctions.scala (hunk 1 above)
Comment (mkearns-ccri) [reply geomesa-spark/geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/jts/udf/GeomesaPysparkFunctions.scala:55]: yep, it's just an alias for the other function

**Comment by elahrvivaz** [comment]
sorry i missed that this was against 2.4.x - since we're following semantic versioning, we can't merge this directly. could you re-open a PR against main? thanks!

**Pull request closed without merging**